unit-tests/src/test/java/org/eclipse/collections/impl/block/procedure/Procedures2Test.java

```diff
@@ -39,6 +42,17 @@ public void throwing()
                 }).value(null, null));
     }
 
+    @Test
+    public void throwingWithSuccessfulCompletion()
+    {
+        MutableList<IntIntPair> list = Lists.mutable.empty();
+        Procedures2.throwing(
+                (ThrowingProcedure2<Integer, Integer>) (object, parameter) -> list.add(
```

Comment:
Is the cast necessary?

Reply:
Yes.
Removing it causes:
14:55:35:489 [ERROR] /unit-tests/src/test/java/org/eclipse/collections/impl/block/procedure/Procedures2Test.java:[51,51] error: cannot find symbol
  symbol:   method intValue()
  location: variable object of type Object
14:55:35:489 [ERROR] /unit-tests/src/test/java/org/eclipse/collections/impl/block/procedure/Procedures2Test.java:[52,41] error: cannot find symbol
  symbol:   method intValue()
  location: variable parameter of type Object
14:55:35:490 [INFO] 2 errors 
14:55:35:490 [INFO] -------------------------------------------------------------

Reply:
Extracted into a separate variable as suggested.